File: cs/Problems/Day7Test.cs
namespace aoc24.Problems;

public class Day7Test
{
    private readonly Day7 sut = new();

    [Theory, InlineData(3749)]
    public void TestSet_ShouldYield_Result(long expected)
    {
        var input = InputReader.ReadProblemInput("day7_1");
        var result = sut.Solve(input);

        Assert.Equal(expected, result);
    }

    [Theory, InlineData(1298103531759)]
    public void FullSet_ShouldYield_Result(long expected)
    {
        var input = InputReader.ReadProblemInput("day7_2");
        var result = sut.Solve(input);

        Assert.Equal(expected, result);
    }
}

File: go/problems/day7_test.go
package problems

import "testing"

func TestDay7_TestSet_YieldsExpectedResult(t *testing.T) {
	TestAocProblem(t, Day7{}, "day7_1", 3749)
}

func TestDay7_FullSet_YieldsExpectedResult(t *testing.T) {
	TestAocProblem(t, Day7{}, "day7_2", 1298103531759)
}
